company-yasnippet.el

```diff
@@ -107,10 +107,15 @@ It has to accept one argument: the snippet's name.")
 
 (defun company-yasnippet--doc (arg)
   (let ((template (get-text-property 0 'yas-template arg))
-        (mode major-mode))
+        (mode major-mode)
+        (file-name (buffer-file-name)))
     (with-current-buffer (company-doc-buffer)
+      (setq-local buffer-file-name file-name)
```

Comment:
Do not set the same file name to both buffers. Should use `let` or set back before returning.

Reply:
`let` would be better, yes.